src/osd/ECCommon.h

```diff
@@ -238,15 +238,20 @@ struct ECCommon {
     bool fast_read,
     GenContextURef<ec_extents_t &&> &&func) = 0;
 
+  struct shard_read_t {
```

Comment:
pls also add an fmt formatter (and maybe consider using that formatter from the operator<<)

Reply:
I have moved most of this into a different PR - but will address these comments there. https://github.com/ceph/ceph/pull/59729